web-api/src/personalization/routes.rs

```diff
@@ -111,19 +119,28 @@ pub(crate) async fn update_interactions(
         .iter()
         .map(|(document_id, _)| document_id)
         .collect_vec();
-    storage::Interaction::update_interactions(storage, user_id, &document_ids, |context| {
-        match document_id_to_interaction_type[&context.document.id] {
-            UserInteractionType::Positive => {
-                for tag in &context.document.tags {
-                    *context.tag_weight_diff
+    storage::Interaction::update_interactions(
+        storage,
+        user_id,
+        &document_ids,
+        |context| {
+            match document_id_to_interaction_type[&context.document.id] {
+                UserInteractionType::Positive => {
+                    for tag in &context.document.tags {
+                        *context.tag_weight_diff
                             .get_mut(tag)
                             .unwrap(/* update_interactions assures all tags are given */) += 1;
-                }
-                coi.log_positive_user_reaction(context.positive_cois, &context.document.embedding)
+                    }
+                    coi.log_positive_user_reaction(
+                        context.positive_cois,
+                        &context.document.embedding,
+                    )
                     .clone()
+                }
             }
-        }
-    })
+        },
+        store_user_history,
```

Comment:
:art: Maybe we should move it before the closure, I'm not sure if it's just because of the diff, but it's quite easy to miss.